### ISSUE TITLE
Should SSP modules require SSP?

### ISSUE THREAD

**Issue by MaximilianKresse** [simplesamlphp/composer-module-installer#13]
First of see https://github.com/simplesamlphp/composer-module-installer/pull/11#issuecomment-561555058

The error was triggered because my module/theme has a composer require statement. I think the error could also be triggered if another package requires SSP and some default modules. 

In my opinion every SSP module MUST depend on SSP directly because it uses code of SSP and/or depends on specific directory structure. SSP is clearly a dependency of SSP modules and should be handled like one.

But I got told that my module shouldn't require SSP. 

Whats speaks against it? Why should modules/themes not be dependent on SSP?
The only reason I know is to prevent circular dependencies but this problem already exists because SSP requires all the modules every module requires the module install and the module installer requires SSP - a perfect circular dependency :-)

**Comment by MaximilianKresse** [comment]
ping?

**Comment by jaimeperez** [comment]
Hi @MaximilianKresse!

You are right, SSP is a dependency for modules, themes, etc. However, you don't need to specify SSP as a **direct** dependency of your module. The only dependency that's needed is `simplesamlphp/composer-module-installer`, which in turn depends on SSP as well. That should be all that's needed for a module to work.

That said, you are also right about circular dependencies. We ended in that situation when we externalised most modules, meanwhile we were including them by default with every stable release. That's indeed a circular dependency, but it will go away with 2.0. There aren't many temporary workarounds for that situation, except for removing the dependency on the modules from SSP and changing it to be a suggestion instead. That could be doable.
